scipy/optimize/_basinhopping.py

```diff
@@ -122,8 +130,12 @@ def _monte_carlo_step(self):
         # steps are not sufficient.
         accept = True
         for test in self.accept_tests:
-            testres = test(f_new=energy_after_quench, x_new=x_after_quench,
-                           f_old=self.energy, x_old=self.x)
+            if inspect.signature(test) == _new_accept_test_signature:
+                testres = test(res_new=minres, res_old=self.incumbent_minres)
+            else:
+                testres = test(f_new=energy_after_quench, x_new=x_after_quench,
+                               f_old=self.energy, x_old=self.x)
```

Comment:
For latter when we decide to move forward with all optimisers, we should raise a warning here inviting to move to the new API.

Reply:
Yup; we can make this public in a separate PR if desired, and at that time we can raise warnings. Currently, it just refactors how Metropolis works (privately).